app/javascript/components/Product/CardGrid.tsx

```diff
@@ -41,8 +41,13 @@ export const useSearchReducer = (initial: Omit<State, "offset">) => {
   const [state, dispatch] = React.useReducer(
     (state: State, action: Action) => {
       switch (action.type) {
-        case "set-params":
-          return { params: action.params, results: null, offset: action.params.from };
+        case "set-params": {
+          const params = {
+            ...action.params,
+            taxonomy: action.params.taxonomy === "discover" ? undefined : action.params.taxonomy,
```

Comment:
This is awkward but necessary because:

- We assume `/xxxx` paths are all taxonomy paths
- `/discover` is now discover index

Thus we must manually exclude this as a taxonomy otherwise we'd try to render this as one (e.g. trying to load recommendations for an invalid "discover" taxonomy).